chemprop/models/model.py

```diff
@@ -212,21 +217,33 @@ def predict_step(self, batch: TrainingBatch, batch_idx: int, dataloader_idx: int
         return self(bmg, X_vd, X_d)
 
     def configure_optimizers(self):
-        opt = optim.Adam(self.parameters(), self.init_lr)
-
-        lr_sched = NoamLR(
-            opt,
-            self.warmup_epochs,
-            self.trainer.max_epochs,
-            self.trainer.estimated_stepping_batches // self.trainer.max_epochs,
-            self.init_lr,
-            self.max_lr,
-            self.final_lr,
+        no_decay_param_name = ["bias", "bn.weight"]
+        decayed_params = dict(params=[], weight_decay=self.weight_decay)
+        nondecayed_params = dict(params=[], weight_decay=0.0)
+        for name, parameter in self.named_parameters():
+            if any(nd in name for nd in no_decay_param_name):
+                nondecayed_params["params"].append(parameter)
+            else:
+                decayed_params["params"].append(parameter)
+        param_groups = [decayed_params, nondecayed_params]
+
+        opt = optim.AdamW(param_groups, self.init_lr)
+        steps_per_epoch = self.trainer.num_training_batches
```

Comment:
```suggestion
        steps_per_epoch = self.trainer.estimated_stepping_batches // self.trainer.max_epochs
```
`num_training_batches` doesn't work in `configure_optimizers`...
https://github.com/Lightning-AI/pytorch-lightning/issues/16060

Reply:
Does this need to be changed? I tested your branch and the training worked for me.

Reply:
I have solved this issue by the other method. Specifically, you can look at the L217-220.